### ISSUE TITLE
Get reported issues into OTRS

### ISSUE THREAD

**Issue by pnorman** [openstreetmap/openstreetmap-website#1903]
We've merged the moderator branch, so people can now report issues that need to make it to the DWG. People with the moderator flag (DWG and admins) can see the issues.

It was raised [in review](https://github.com/openstreetmap/openstreetmap-website/pull/1576#issuecomment-379226550) that we needed to look at the interface between getting a report and the DWG being able to act.

Right now that is lacking. Reporting an issue does **not** get it into OTRS, which is where we do our work. This means I need to either reply to the issue telling them to contact the Data Working Group by email instead of out of band comms, copy it myself, or ignore the issue. The first is unhelpful, the second is not going to happen based on experience, and the third is unhelpful and what's going to happen in practice.

Until we figure out how we get issues from a report into a form we can act on, can we disable reporting of stuff that normally goes to the DWG?

**Comment by gravitystorm** [comment]
I think we need to discuss and agree the different scopes of "moderators" and "DWG". I've previously thought that there would be a lot of moderation work (e.g. deleting spam diary entries) that don't need to have any DWG involvement at all.

But is this how other people see it? Or is there actually a 1:1 correlation between accounts with the moderator flag and DWG members? Unfortunately there's no public list of moderators (or admins, maybe I should add this to the code) so it's hard for me to figure out the current situation, never mind plan for the future. I don't know what percentage of "moderator issues" are currently being dealt with by DWG members, and whether they feel like the two concepts are interchangeable.

There's also the case of escalating issues. It's currently possible for issues to be dealt with by moderators and escalated to admins if necessary. Do we need to separate out the concepts of "stuff DWG deals with" from "the rest of the issues assigned to the moderators"?

Again, it's hard for me to deal with this, since I don't actually get to see what's in the issues list and how it's being dealt with! I'm neither an admin nor a moderator on the osm.org deployment, so I need to rely on feedback from everyone else involved.

**Comment by SomeoneElseOSM** [comment]
> Or is there actually a 1:1 correlation between accounts with the moderator flag and DWG members?

Essentially, yes.  The list of moderators (visible to moderators via the "Select Last Updated By" search criteria on the issues list) matches "who we are" on https://wiki.osmfoundation.org/wiki/Data_Working_Group apart from a couple of "special accounts" such as https://www.openstreetmap.org/user/Redaction%20Account%20Bugfix and https://www.openstreetmap.org/user/OSMF%20Data%20Working%20Group .

**Comment by SomeoneElseOSM** [comment]
For info, here's the workflow I'm currently using for OSM issues (I'm a moderator, so see complaints about notes and some complaints about users in the issues list):

- Look at the names involved and see if it corresponds to an existing OTRS ticket.  If it does, add details of the issue to the OTRS ticket as either "pending" or "phone call incoming".  If it doesn't, create a new ticket.
- See if there are any other related issues (often person A complains about person B and person B also complains about person A) and merge those too.
- Send an OSM PM to whoever's complained saying "thank you for your report" and asking for any more information, and copy that PM to OTRS as a note.
- Add a comment to the the OSM issue linking to the OTRS ticket and resolve the OSM issue to avoid someone else trying to pick it up later.
- If there's any immediate action required (block, revert, etc.) do that, and PM any other involved parties if necessary.
- Wait for replies, handling the ticket in OTRS as normal, with the exception that instead of by email messages are sent as OSM PMs and copied to the ticket as notes.
- When all complete, close the ticket in OTRS

**Comment by gravitystorm** [comment]
> The list of moderators (visible to moderators via the "Select Last Updated By" search criteria on the issues list) matches "who we are" on https://wiki.osmfoundation.org/wiki/Data_Working_Group

Thanks.

> For info, here's the workflow I'm currently using for OSM issues

Do you do all this for diary entry spam too? Or something different?

**Comment by SomeoneElseOSM** [comment]
> Do you do all this for diary entry spam too? Or something different?

Diary spam goes to admins rather than moderators, so no.

"notes spam" does come to moderators, and the answer for that is "sometimes".  If it's a known spammer (usually not a known account but a familiar type of spam from a group that hasn't responded in the past) and there's no further user interaction needed then I just hide the note.

If there is some user interaction needed (e.g. when people report as spam things that aren't) I tend to create an OTRS ticket, PM the person who reported it, copy that PM into OTRS, resolve the OSM issue and close the OTRS ticket.

Also to be clear different DWG members are using different workflows - I notice that mavl tends to add a comment to say "I'm working on this" rather than resolving immediately, as I have been doing.

**Comment by mikelmaron** [comment]
@SomeoneElseOSM what does the workflow look like for non-data issues, like reports of abuse or threats?

Are there any quick summary stats on the kinds of issues and what type of content was reported?

**Comment by SomeoneElseOSM** [comment]
> what does the workflow look like for non-data issues, like reports of abuse or threats?

Typically the same as I outlined, except in the case of abuse or a threat there'd be an immediate action (remove/hide the abuse if appropriate, and take action against the abusing user as required).

> Are there any quick summary stats on the kinds of issues and what type of content was reported?

Most of what I've seen come through https://www.openstreetmap.org/issues so far has been categorised as "spam", "other", or "vandal".  I don't believe that I've seen an example of "threat" or "obscene/offensive" via that mechanism yet.  The last "obscene/offensive" issue that I dealt with was a series of offensive changeset discussion comments that I happened to notice because I'd subscribed to the changeset because it was in a disputed geographical area; that was about a month ago.  In that case I created the DWG ticket myself to capture the actions; it was never reported to us as a problem because I hid the offensive comments before they were widely noticed.

**Comment by mboeringa** [comment]
> I've previously thought that there would be a lot of moderation work (e.g. deleting spam diary entries) that don't need to have any DWG involvement at all.

@gravitystorm ,

This may have been suggested before, but would it be an option to disallow creating diary entries unless a user has made an X number of OpenStreetMap edits? 

Recently, there has been quite a flurry of spam diary entries with most if not all users having zero OpenStreetMap edits. Obviously, requiring edits to be made, might reduce the number of spam diary entries (at the slight risk of increased junk edits, but I have seen only a few cases of combined junk edit + diary spam, the majority of spam is with 0 edits).

**Comment by dieterdreist** [comment]
Am Mi., 27. Feb. 2019 um 10:27 Uhr schrieb mboeringa <
notifications@github.com>:

> Recently, there has been quite a flurry of spam diary entries with most if
> not all users having zero OpenStreetMap edits. Obviously, requiring edits
> to be made, might reduce the number of spam diary entries (at the slight
> risk of increased junk edits, but I have seen only a few cases of combined
> junk edit + diary spam, the majority of spam is with 0 edits).



I agree we could restrict the creation of diary entries to accounts which
have at least 1 (or 23?) map edit(s). There could be an exception for OSMF
board members if required.

**Comment by tomhughes** [comment]
The number of edits is already part of the scoring that can trigger automatic suspension - as in less edits means you are more likely to trigger it.

It's not uncommon for people to post diaries without editing though so I don't really want to just ban it.

**Comment by tomhughes** [comment]
By the way as the person dealing with the spam reports I have a pretty good idea how many we're getting and it's really not that many though obviously more than I like.

**Comment by tomhughes** [comment]
The numbers since the start of February look like this:

* 81 posts by active accounts
* 51 posts by suspended accounts - this will be automatic blocks
* 57 posts by deleted accounts - this will be manual blocks

So ignoring the automatic blocks where, in most cases, nobody will ever see the post it's about 40% spam most of which will only last a few hours before being killed.

Obviously it's not ideal but it's hard to do anything about it without impacting on genuine users - we already get the occasional case of a genuine post (usually by a new user with no edits) getting blocked by the automatic blocker.

**Comment by mboeringa** [comment]
> It's not uncommon for people to post diaries without editing though so I don't really want to just ban it.

Even if that is the case, requiring them to perform at least a single edit, shouldn't be a tough requirement. 

If these users are dedicated enough to OpenStreetMap to make a genuinely useful OpenStreetMap related diary post, they should find it no issue to make some edits as well.

Today, we at least seem to have a "bad hair day" in terms of spam, this is the latest one after having reported another 4 this morning:
https://www.openstreetmap.org/user/Hairsmarket/diary/47813

**Comment by SomeoneElseOSM** [comment]
On the original issue "Get reported issues into OTRS" - it's got a bit easier since 2018 because Anton from the DWG has written a Firefox add-on "OSM DWG Helper" that converts issues into OTRS tickets (by rekeying it, automatically).  Once things are out of the osm.org world and into OTRS we can reply to the reporter via an OTRS -> osm.org message gateway set up by another DWG member.

It's not exactly "integration", but it's a lot better than things were 4 years ago.
